cql3/restrictions/statement_restrictions.cc

```diff
@@ -1286,16 +1286,27 @@ bool statement_restrictions::need_filtering() const {
         // clustering restrictions.  Therefore, a continuous clustering range is guaranteed.
         return false;
     }
-    if (!_clustering_columns_restrictions->needs_filtering(*_schema)) { // Guaranteed continuous clustering range.
-        return false;
-    }
-    // Now we know there are some clustering-column restrictions that are out-of-order or not EQ.  A naive base-table
-    // query must be filtered.  What about an index-table query?  That can only avoid filtering if there is exactly one
-    // EQ supported by an index.
-    return !(_clustering_columns_restrictions->size() == 1 && _has_queriable_ck_index);
 
-    // TODO: it is also possible to avoid filtering here if a non-empty CK prefix is specified and token_known, plus
-    // there's exactly one out-of-order-but-index-supported clustering-column restriction.
+    if (_has_queriable_ck_index && _uses_secondary_indexing) {
+        // In cases where we use an index, clustering column restrictions might cause the need for filtering.
+        // TODO: This is overly conservative, there are some cases when this returns true but filtering
+        // is not needed. Because of that the user will now sometimes be forced to add ALLOW FILTERING
```

Comment:
Because of this change we now ask users to specify `ALLOW FILTERING` in cases where it is not needed. This function might not be explicitly responsible for requiring `ALLOW FILTERING`, but it clearly influences it. Maybe it's worth noting there that in these cases we ask to `ALLOW FILTERING` because of this code?

Maybe something like
```
Because of that the database will sometimes perform filtering when it's not actually needed and require the user to specify `ALLOW FILTERING'.
Query performance shouldn't worsen much, at most we will filter rows that are all correct.
```

Reply:
> This function might not be explicitly responsible for requiring `ALLOW FILTERING`, but it clearly influences it.

... today.  As a general rule, no function should presume what its callers do with it.

> Maybe it's worth noting there that in these cases we ask to `ALLOW FILTERING` because of this code?

If so, that note belongs in the commit message, not inside the function.

Reply:
ok I removed all `ALLOW FILTERING` from the comment